# Numpy Select

## numpy.select(condlist, choicelist, default=0)

## Return an array drawn from elements in choicelist, depending on conditions.



## Parameters
###     condlist: list of bool ndarrays
###     The list of conditions which determine from which array in choicelist the output elements are taken. When multiple conditions are satisfied, the first one encountered in condlist is used.

###     choicelist: list of ndarrays
###     The list of arrays from which the output elements are taken. It has to be of the same length as condlist.

###      default: scalar, optional
###      The element inserted in output when all conditions evaluate to False.
## Returns
### outputndarray
### The output at position m is the m-th element of the array in choicelist where the m-th element of the corresponding array in condlist is True.

## Now, Lets see it in an example:

In [6]:
import numpy

x = numpy.arange(10)

condlist = [x<5,x>8]

choicelist = [x, x**3]

numpy.select(condlist,choicelist)

array([  0,   1,   2,   3,   4,   0,   0,   0,   0, 729])

# x here is an array from 0 to 9 (we get this array via numpy.arrange(10)).So, basically the output gives us an array. As you can see, we have 0,1,2,3,4 and after that comes zeros and 729 at the end. The reason is that in our condition we say that when x is less than 5 then it should be in a form of x and if x>8 then it should be shown as x**3 (calculate 9**3 and you will get 729). 

# Now lets set default as 100 and see what will happen.

In [8]:
import numpy

x = numpy.arange(10)

condlist = [x<5,x>8]

choicelist = [x, x**3]

numpy.select(condlist,choicelist,100)

array([  0,   1,   2,   3,   4, 100, 100, 100, 100, 729])

# As you can see, all zeros after 4 changed to 100. The reason is that we set number to 100 whenever the condition is not satisfied (in our case numbers between 4 and 8 are those numbers). In an example before it was zeros as we didnt set the default.